lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -160,6 +169,30 @@ def scopes_match?(token_scopes, param_scopes, app_scopes)
           )
       end
 
+      # Checks whether the token custom attribute values match the custom
+      # attributes from the parameters.
+      #
+      # @param token [Doorkeeper::AccessToken]
+      #   The access token whose custom attributes are being compared
+      #   to the param_attributes.
+      #
+      # @param param_attributes [Hash]
+      #   A hash of the attributes for which we want to determine whether
+      #   the token's custom attributes match.
+      #
+      # @return [Boolean] true if the token's custom attribute values
+      #   match those in the param_attributes, or if both are empty/blank.
+      #   False otherwise.
+      def custom_attributes_match?(token, param_attributes)
```

Comment:
Can we rename `param_attributes` to `custom_attributes`? These do not necessarily come from the request, but from the authorization code record as well, and it's a bit confusing.